### DISCUSSION TITLE
foundry test

### DISCUSSION THREAD

**Discussion by sebghatullahmalikzay** [Cyfrin/foundry-full-course-cu#492] [General]
hi everyone what is the work of the foundry I saw the video but I don't know the work of the foundry what is specifically without testing the code in foundry we also know this is correct and also they is in correct what foundry for but we write code two place instead of one place.

is foundry for security of the contract or not?

**Comment by Niferu** [comment]
To be honest I have no idea what you are asking for as your question is formatted in pretty bizzare english. If you are asking what foundry is, it is just framework, which allows you to create, deploy and test your smart contracts.

**Reply by sebghatullahmalikzay** [reply]
thank you

**Comment by theirrationalone** [comment]
Sorry to say, But It's hard to Understand You (You Should Groom Yourself)... But according to my Guess You might be asking... What Foundry actually is??....

It's just a Suit (solidity as Script)... A testing Environment with huge amount of elegant tools(i.e. Forge, Anvil, Cast, Chisel, In Abstraction. ) in hand that help to find vulnerabilities in Smart Contracts and Foundry as a [Book](https://book.getfoundry.sh) suggests tons of tips(TOTs) to make our Smart Contract Powerful, Ethical, Immutable, Unbreakable, Un-hackable, Concise, Precise, Algorithmically Decentralized, Transparent. 

Thank You!

**Reply by sebghatullahmalikzay** [reply]
thank You

**Reply by theirrationalone** [reply]
You're most Welcome, However, Please close the Discussion! 😉
